src/OAuth1/Provider/Atlassian.php

```diff
@@ -0,0 +1,150 @@
+use SocialConnect\Provider\Exception\InvalidResponse;
+use SocialConnect\OAuth1\AbstractProvider;
+use SocialConnect\Common\Entity\User;
+use SocialConnect\Common\Hydrator\ObjectMap;
+use SocialConnect\Provider\Session\SessionInterface;
+
+class Atlassian extends AbstractProvider
+{
+    private $baseUri;
+    /**
+     * {@inheritdoc}
+     */
+    public function getBaseUri()
+    {
+        return $this->baseUri;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAuthorizeUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/authorize';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequestTokenUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/request-token';
+    }
+
+    /**
+     * @return string
+     */
+    public function getRequestTokenAccessUri()
+    {
+        return $this->getBaseUri() . '/plugins/servlet/oauth/access-token';
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'atlassian';
+    }
+
+    /**
+     * Atlassian constructor.
+     *
+     * Required configuration parameters are:
+     *
+     * * base_uri The base URI of your self-hosted atlassian product
+     * * private_key The path to the private key file used for signing.
+     *
+     * @param \SocialConnect\Common\Http\Client\ClientInterface $httpClient
+     * @param \SocialConnect\Provider\Session\SessionInterface  $session
+     * @param \SocialConnect\Provider\Consumer           $consumer
+     * @param array                                             $parameters
+     */
+    public function __construct(ClientInterface $httpClient, SessionInterface $session, Consumer $consumer, array $parameters)
+    {
+        if (! isset($parameters['baseUri'])) {
+            throw new \InvalidArgumentException('There is no "baseUri" given in the configuration');
+        }
+        $this->baseUri = $parameters['baseUri'];
+        if (($lastSlash = strrpos($this->baseUri, '/')) == strlen($this->baseUri) - 1) {
+            $this->baseUri = substr($this->baseUri, 0, $lastSlash);
+        }
+
+        parent::__construct($httpClient, $session, $consumer, $parameters);
+
+        // needs to be set after calling the parent constructor as there the
+        // signature is set as well.
+        $this->signature = new MethodRSASHA1($consumer->getSecret());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentity(AccessTokenInterface $accessToken)
+    {
+        $this->consumerToken = $accessToken;
+
+        $parameters = [
+            'oauth_consumer_key' => $this->consumer->getKey(),
+            'oauth_token' => $this->consumerToken->getToken(),
+        ];
+
+        $response = $this->oauthRequest(
+            $this->getBaseUri() . '/rest/prototype/1/user/current',
+            Client::GET,
+            $parameters
+        );
+        while ($response->hasHeader('Location')) {
```

Comment:
plank line before while
and btw, this while can be an infinity loop
if you are sure that attlasin can lag, maybe add retryCount, and retryMax? Thank